### PR TITLE
[PDI-18290] - Can't execute a transformation with an ampersand in the…

### PR DESCRIPTION
**Pull request by bcostahitachivantara**
… name on slave server

    * Use JAXB to escape special XML characters
    * Modify unit test to include xml prolog and remove new lines

### CODE REVIEW

**Comment by buildguy** [comment]
### :gear: Build finished in 1m 34s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl plugins/engine-configuration/impl
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|52|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6911/1/testReport)|
|----|


:information_source:  _This is an automatic message_

File: plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/JobScheduleParam.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  private static final long serialVersionUID = -4214459740606299083L;
+
+  private static final SimpleDateFormat isodatetime = new SimpleDateFormat( "yyyy-MM-dd'T'HH:mm:ss.SSSZZZ" );
+    //$NON-NLS-1$
+
+  String name;
+
+  String type;
+
+  ArrayList<String> stringValue = new ArrayList<String>();
```
Comment (pentaho-lmartins) [inline plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/JobScheduleParam.java]: Can you write: `List<String> stringValue = new ArrayList<>();`

File: plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/JobScheduleParam.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  public ArrayList<String> getStringValue() {
+    return stringValue;
+  }
+
+  public void setStringValue( ArrayList<String> value ) {
+    this.stringValue = value;
+  }
+
+}
```
Comment (pentaho-lmartins) [inline plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/JobScheduleParam.java]: No unit tests for this new class? I'd like to see 100% coverage!

File: plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/JobScheduleRequest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+@XmlRootElement( name = "jobScheduleRequest" )
+@XmlAccessorType( XmlAccessType.FIELD )
+@XmlType( propOrder = { "inputFile", "jobParameters", "pdiParameters" } )
+public class JobScheduleRequest implements Serializable {
+
+  private static final long serialVersionUID = -485489832281790257L;
+
+  String inputFile;
+
+  ArrayList<JobScheduleParam> jobParameters = new ArrayList<>();
```
Comment (pentaho-lmartins) [inline plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/JobScheduleRequest.java]: Can you declare this a `List`?

File: plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/SchedulerRequest.java [hunk 4]
```diff
@@ -129,77 +135,87 @@ private String getFullPath( AbstractMeta meta ) {
...
-    sb.append( "<jobParameters>\n" );
-    sb.append( "<name>" ).append( "clearLog" ).append( "</name>\n" );
-    sb.append( "<type>" ).append( "string" ).append( "</type>\n" );
-    sb.append( "<stringValue>" ).append( meta.isClearingLog() ).append( "</stringValue>\n" );
-    sb.append( "</jobParameters>\n" );
+    JobScheduleParam jobScheduleParamClearingLog = new JobScheduleParam();
+    jobScheduleParamClearingLog.setName( "clearLog" );
+    jobScheduleParamClearingLog.setType( "string" );
+    jobScheduleParamClearingLog
+      .setStringValue( new ArrayList<>( Arrays.asList( String.valueOf( meta.isClearingLog() ) ) ) );
```
Comment (pentaho-lmartins) [inline plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/SchedulerRequest.java]: `new ArrayList<>( Arrays.asList(`.... is this really necessary? See comments above about using `List`.

File: plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/SchedulerRequest.java [hunk 5]
```diff
@@ -129,77 +135,87 @@ private String getFullPath( AbstractMeta meta ) {
...
+    // Marshal object to string
+    StringWriter sw = new StringWriter();
+    String jobScheduleRequestString = null;
+    try {
+      JAXBContext jc = JAXBContext.newInstance( JobScheduleRequest.class );
+      Marshaller marshaller = jc.createMarshaller();
+      marshaller.marshal( jobScheduleRequest, sw );
+      jobScheduleRequestString = sw.toString();
+    } catch ( JAXBException e ) {
+      e.printStackTrace();
```
Comment (pentaho-lmartins) [inline plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/SchedulerRequest.java]: Use a logger instead....

**Comment by buildguy** [comment]
### :gear: Build finished in 1m 39s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl plugins/engine-configuration/impl
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|52|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6911/2/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 1m 37s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl plugins/engine-configuration/impl
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|52|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6911/3/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 1m 32s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl plugins/engine-configuration/impl
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|64|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6911/4/testReport)|
|----|


:information_source:  _This is an automatic message_

**Pull request merged**